[Bug]: Rolldown panics when enabling `transform.decorators.legacy: true` without installing `@oxc-project/runtime`

### Reproduction link or steps

https://stackblitz.com/edit/rolldown-rolldown-starter-stackblitz-eqrhhsvw

### What is expected?

No panic.

### What is actually happening?

Panic.

```
 ERROR  Build failed with 1 error:                                                     

[UNHANDLEABLE_ERROR] Error: Something went wrong inside rolldown, please report this problem at https://github.com/rolldown/rolldown/issues.
Cannot find module '@oxc-project/runtime/helpers/decorate'
```

### System Info

```Shell
.
```

### Any additional comments?

After run `pnpm add @oxc-project/runtime`, it works.

_No response_

--- a/knip.jsonc
+++ b/knip.jsonc
@@ -27,7 +27,7 @@
       "ignoreDependencies": ["@babel/preset-.+", "react", "react-dom"],
     },
     "packages/browser": {
-      "ignoreDependencies": ["@napi-rs/wasm-runtime"],
+      "ignoreDependencies": ["@napi-rs/wasm-runtime", "@oxc-project/runtime"],
     },
     "packages/rolldown": {
       "entry": [

--- a/packages/browser/package.json
+++ b/packages/browser/package.json
@@ -62,6 +62,7 @@
     "preinstall": "npx only-allow pnpm"
   },
   "dependencies": {
-    "@napi-rs/wasm-runtime": "catalog:"
+    "@napi-rs/wasm-runtime": "catalog:",
+    "@oxc-project/runtime": "catalog:"
   }
 }

